stellar-multitenant/pkg/tenant/manager.go

```diff
@@ -53,32 +55,49 @@ func (m *Manager) GetDSNForTenant(ctx context.Context, tenantName string) (strin
 	return router.GetDSNForTenant(dataSourceName, tenantName)
 }
 
+func (m *Manager) GetDSNForTenantByID(ctx context.Context, id string) (string, error) {
+	t, err := m.GetTenantByID(ctx, id, nil)
+	if err != nil {
+		return "", err
+	}
+
+	return m.GetDSNForTenant(ctx, t.Name)
+}
+
 var selectQuery string = `
 	SELECT 
 		*
 	FROM
 		tenants t
-	%s
 `
 
 // GetAllTenants returns all tenants in the database.
-func (m *Manager) GetAllTenants(ctx context.Context) ([]Tenant, error) {
-	tnts := []Tenant{}
-
-	query := fmt.Sprintf(selectQuery, "ORDER BY t.name ASC")
+func (m *Manager) GetAllTenants(ctx context.Context, queryParams *QueryParams) ([]Tenant, error) {
+	if queryParams == nil {
+		queryParams = &QueryParams{}
+	}
 
-	err := m.db.SelectContext(ctx, &tnts, query)
+	tnts := []Tenant{}
+	query, params := m.newManagerQuery(selectQuery, queryParams)
+	err := m.db.SelectContext(ctx, &tnts, query, params...)
 	if err != nil {
 		return nil, fmt.Errorf("getting all tenants: %w", err)
 	}
 
 	return tnts, nil
 }
 
-func (m *Manager) GetTenantByID(ctx context.Context, id string) (*Tenant, error) {
-	q := fmt.Sprintf(selectQuery, "WHERE t.id = $1")
+func (m *Manager) GetTenantByID(ctx context.Context, id string, queryParams *QueryParams) (*Tenant, error) {
```

Comment:
I like this refactor but there are a couple of adjustments needed I believe. 

I don't see any usages in the production code (non-test code) for the **queryParams** here. But the other issue is that we're not filtering out deactivated tenants by default. 

I believe we should change this and most methods in the manager to **filter out deactivated tenants by default** and create specific methods for fetching deactivated tenants **if needed** (can also use the generic `GetTenant` for special cases). 

In your next story, we will also need to change these methods to filter out deleted tenants by default. 

This is an example implementation of what I'm talking about (I didn't test this code) 
```golang
// GetTenant generic method that fetches a tenant based on queryParams
func (m *Manager) GetTenant(ctx context.Context, queryParams *QueryParams) (*Tenant, error) {
	if queryParams == nil {
		queryParams = &QueryParams{}
	}

	var t Tenant
	query, params := m.newManagerQuery(selectQuery, queryParams)
	err := m.db.GetContext(ctx, &t, query, params...)
	if err != nil {
		if errors.Is(err, sql.ErrNoRows) {
			return nil, ErrTenantDoesNotExist
		}
		return nil, fmt.Errorf("getting tenant: %w", err)
	}

	return &t, nil
}

func (m *Manager) GetTenantByID(ctx context.Context, id string) (*Tenant, error) {
	queryParams := &QueryParams{
		Filters: map[FilterKey]interface{}{
			FilterKeyID: id,
                         FilterKeyOutStatus: DeactivatedTenantStatus,
		},
	}
	
	return m.GetTenant(ctx, queryParams)
}

func (m *Manager) GetDeactivatedTenantByID(ctx context.Context, id string) (*Tenant, error) {
	queryParams := &QueryParams{
		Filters: map[FilterKey]interface{}{
			FilterKeyID: id,
			FilterKeyStatus: DeactivatedTenantStatus,
		},
	}
	
	return m.GetTenant(ctx, queryParams)
}

func (m *Manager) GetTenantByName(ctx context.Context, name string) (*Tenant, error) {
	queryParams := &QueryParams{
		Filters: map[FilterKey]interface{}{
			FilterKeyName: name,
                         FilterKeyOutStatus: DeactivatedTenantStatus,
		},
	}
	
	return m.GetTenant(ctx, queryParams)
}

... etc. 
```